homeassistant/components/switch/rainmachine.py

```diff
@@ -0,0 +1,308 @@
+"""Implements a RainMachine sprinkler controller for Home Assistant."""
+
+import asyncio
+from datetime import timedelta
+from logging import getLogger
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import (ATTR_ATTRIBUTION, ATTR_DEVICE_CLASS,
+                                 CONF_EMAIL, CONF_IP_ADDRESS, CONF_PASSWORD)
+from homeassistant.util import Throttle
+
+_LOGGER = getLogger(__name__)
+REQUIREMENTS = ['regenmaschine==0.3.2']
+
+ATTR_CYCLES = 'cycles'
+ATTR_TOTAL_DURATION = 'total_duration'
+
+CONF_HIDE_DISABLED_ENTITIES = 'hide_disabled_entities'
+CONF_ZONE_RUN_TIME = 'zone_run_time'
+
+DEFAULT_ZONE_RUN_SECONDS = 60 * 10
+
+MIN_SCAN_TIME_LOCAL = timedelta(seconds=1)
+MIN_SCAN_TIME_REMOTE = timedelta(seconds=5)
+MIN_SCAN_TIME_FORCED = timedelta(milliseconds=100)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_IP_ADDRESS):
```

Comment:
This is great, but I'm confused on how to utilize this? Without any extra changes, I get exceptions about my `ip_address` and `email_address` variables (set in `async_setup_platform()`) being functions.